### PR TITLE
add version field to Entry type

### PR DESCRIPTION
**Pull request by michaelbeam**

### CODE REVIEW

File: entry.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 )
 
 type Entry struct {
+	Version uint8    `json:"version"`
```
Comment (Emyrk) [inline entry.go]: The json marshalling is overridden in a custom MarshalJson function. The version is not included there, or in the unmarshal json

**Comment by PaulBernier** [comment]
@michaelbeam any change to address Steven feedback? If not we will close this PR, thanks.

**Comment by WhoSoup** [comment]
> @michaelbeam any chance to address Steven feedback? If not we will close this PR, thanks.

The API doesn't send the entry version number along, it is only included when decoding the raw entry data. Adding the version field would mean that it is 0 (default value) for entries returned from the API and 0 (actual value) when decoding raw entries, which are coincidentally the same number.

Options here are:
1. Making a change on the factomd api side to include "version" for entries
2. Not including the version in the entry because factom currently only has one entry version and that's unlikely to change any time soon, making this useless information

I'm leaning toward 2. and we can introduce the version number here when the factom protocol actually has more than one entry version.

**Comment by PaulBernier** [comment]
@WhoSoup agreed. Closing this PR.

**Pull request closed without merging**